cloudinstall/placement/ui/__init__.py

```diff
@@ -165,9 +165,6 @@ def build_widgets(self):
         self.deploy_ok_msg = ("\u2713 All the required OpenStack services are "
                               "placed on a machine, and you can now deploy.")
 
-        # Save a new placements configuration once all required services are
-        # placed.
-        self.placement_controller.save()
```

Comment:
I double checked the history of this file and the placement.save was added when I switched to using the config object to store the placement. It looks like in that particular scenario the placements weren't being saved soon enough. However, with this code the placements file is being written out accordingly and reloaded during re-runs of `edit_placement`. I think it's safe to leave it out for now, at least until your code rework is implemented.